otter/integration/tests/test_convergence.py

```diff
@@ -150,6 +150,55 @@ def wrapper(*args, **kwargs):
         return decorated
 
 
+def tag(*tags):
+    """
+    Decorator that adds tags to a function by setting the property "tags".
+
+    This should be added upstream to Twisted trial.
+    """
+    def decorate(function):
+        function.tags = tags
+        return function
+    return decorate
+
+
+def skip_me(reason):
+    """
+    Decorator that skips a test method or test class by setting the property
+    "skip".  This decorator is not named "skip", because setting "skip" on a
+    module skips the whole tes module.
+
+    This should be added upstream to Twisted trial.
+    """
+    def decorate(function):
+        function.skip = reason
+        return function
+    return decorate
+
+
+def copy_test_methods(from_class, to_class, filter_and_change=None):
+    """
+    Copy test methods (methods that start with `test_*`) from ``from_class`` to
+    ``to_class``.  If a decorator is provided, the test method on the
+    ``to_class`` will first be decorated before being set.
+
+    :param class from_class: The test case to copy from
+    :param class to_class: The test case to copy to
+    :param callable filter_and_change: A function that takes a test name
+        and test method, and returns a tuple of `(name, method)`
+        if the test method should be copied. None else.  This allows the
+        method name to change, the method to be decorated and/or skipped.
+    """
+    for name, attr in from_class.__dict__.items():
```

Comment:
I don't feel strongly about this, but lean towards I `__dict__` because it seems like less work.  :)   `dir` gives all inherited members and all the python magic members, whereas `__dict__` gives only the members defined in the class.  e.g.:

```
>>> class A(object):
...    def a(self):
...       pass
... 
>>> class B(A):
...   def b(self):
...      pass
... 
['__class__', '__delattr__', '__dict__', '__doc__', '__format__', '__getattribute__', '__hash__', '__init__', '__module__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', 'a', 'b']
>>> B.__dict__
dict_proxy({'__module__': '__main__', 'b': <function b at 0x107b61c80>, '__doc__': None})
```

This definitely isn't currently a problem, since this method is written so we can avoid inheritance of test cases anyway, and the function already filters out anything that doesn't start with `test_` and further takes a callable that can customize filtering anyway.  So using `dir` is fine, and not an issue, `__dict__` just seemed simpler (less to think about if there is any test inheritance, and also I don't have to use `getattr`)

I am happy to be overruled if you feel more strongly about it. :D